.github/workflows/ci.yml

```
@@ -40,7 +40,7 @@ jobs:
         run: sudo apt-get update; sudo apt-get install --no-install-recommends libasound2-dev libudev-dev
 
       - name: Build & run doc tests
-        run: cd code-validation && cargo test
+        run: cd learning-code-examples && cargo check --examples && cargo clippy --examples && cargo fmt --check
```

```suggestion
        run: cd learning-code-examples && ./validate_examples.sh
```

It might be worth replacing this long line with a call to `validate_examples.sh`.